src/main/java/evaluation/ParameterSearch.java

```diff
@@ -0,0 +1,210 @@
+package evaluation;
+
+import core.AbstractPlayer;
+import evodef.*;
+import games.GameType;
+import ntbea.*;
+import players.PlayerFactory;
+import players.mcts.MCTSSearchSpace;
+import players.simple.RandomPlayer;
+import utilities.Pair;
+import utilities.StatSummary;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static utilities.Utils.getArg;
+
+public class ParameterSearch {
+
+    public static void main(String[] args) {
+        List<String> argsList = Arrays.asList(args);
+        if (argsList.contains("--help") || argsList.contains("-h")) System.out.println(
+                "The first three arguments must be \n" +
+                        "\t<filename for searchSpace definition>\n" +
+                        "\t<number of NTBEA iterations>\n" +
+                        "\t<game type> \n" +
+                        "Then there are a number of optional arguments:\n" +
+                        "\tnPlayers=      The total number of players in each game (the default is game.Min#players) \n " +
+                        "\tevalGames=     The number of games to run with the best predicted setting to estimate its true value (default is 20% of NTBEA iterations) \n" +
+                        "\topponent=      The filename for the agent used as opponent. Default is a Random player. \n" +
+                        "\tuseThreeTuples If specified then we use 3-tuples as well as 1-, 2- and N-tuples \n" +
+                        "\tkExplore=      The k to use in NTBEA - defaults to 100.0 \n" +
+                        "\thood=          The size of neighbourhood to look at in NTBEA. Default is min(50, |searchSpace|/100) \n" +
+                        "\trepeat=        The number of times NTBEA should be re-run, to find a single best recommendation \n" +
+                        "\tverbose        Will log the results marginalised to each dimension, and the Top 10 best tuples for each run \n"
+        );
+
+        if (argsList.size() < 3)
+            throw new AssertionError("Must specify at least three parameters: searchSpace, NTBEA iterations, game");
+        String searchSpaceFile = args[0];
+        int iterationsPerRun = Integer.parseInt(args[1]);
+        GameType game = GameType.valueOf(args[2]);
+//        if (game != GameType.Dominion)
+//            throw new AssertionError("Only Dominion currently supported");
+        int repeats = getArg(args, "repeat", 1);
+        int evalGames = getArg(args, "evalGames", iterationsPerRun / 5);
+        double kExplore = getArg(args, "kExplore", 100.0);
+        String opponentFile = getArg(args, "opponent", "");
```

Comment:
I wonder if we can have a different argument instead here, as list of players, where each is a string of either the sample players, or a custom class name. Having a whole separate file defining 1 or 2 lines that are just strings really could be avoided in this case I think? 

Or, instead of player name/class, a JSON file could be specified (different parsing for string ending with ".json") which would also contain parameter definition for the players? That would make sense to be in a separate file, and allow flexibility on setting up the evaluation.

Reply:
Done. ParameterSearch and PlayerFactory now support a short-hand notation as (I think) yous suggested. 
At the moment the code only supports a single opponent; so this is straightforward. I do like the idea of having a single config file for more complex experiments/runs that details all the game, player, opponent parameters in one place. But..future enhancement?